brother_printer_fwupd/autodiscover_printer.py

```diff
@@ -3,219 +3,225 @@
 # pylint: disable=C0103
 # pylint: disable=R1723
 
+__all__ = ["PrinterDiscoverer"]
+
 import ipaddress
 import typing
 
 import termcolor
-import zeroconf
-
-from .models import MDNSPrinterInfo
-from .utils import clear_screen, print_debug, print_error
 
-if typing.TYPE_CHECKING:
-    from .models import IPAddress
+try:
+    import zeroconf
+except ImportError:
+    PrinterDiscoverer = None
+else:
```

Comment:
Can we find a solution where we don't have to wrap the entire code in the `else` block?